contracts/voting/Makefile.toml

```diff
@@ -1,3 +1,38 @@
+[plugins.impl.retry_script]
+script = '''
+
+assert ${task.has_script}
+
+taskjson = json_parse ${task.as_json}
+script = set ${taskjson.script}
+
+
+fn retry_script
+  max_retries = set ${1}
+  current_retries = set ${2}
+  script = set ${3}
+  
+  out = exec bash -c ${script}
+  if not eq ${out.code} 0 
+
+    if not eq ${current_retries} ${max_retries}
+      new_current_retries = calc ${current_retries} + 1
+      sleep_time_millis = calc ${new_current_retries} * ${new_current_retries} * 1000
+      println "Retrying .. waiting ${sleep_time_millis} ms before trying again ..."
+      sleep ${sleep_time_millis}
+      println Trying task ${task.name} - try number ${new_current_retries}
+      retry_script ${max_retries} ${new_current_retries} ${script}
+    else
+      println "Max retries reached !"
+      exit 1
+    end
+  else
+  end
+end
+
+retry_script 3 0 ${script}
```

Comment:
This is a custom `cargo-make` plugin implementation of an exponential backoff for retrying transactions. Its based in the `cargo-make` [plugin](https://github.com/sagiegurari/cargo-make#usage-plugins) feature plus, some scripting on [duckscript](https://github.com/sagiegurari/duckscript) :duck: , its native scripting language.

Now we just need to mark the tasks we want to enable the exp retry by adding the `plugin = "retry_script"` attribute to it.


In order to test this, you can clone this branch and adding a dummy task like:

```toml
[tasks.run_retry]
plugin = "retry_script"
script = '''
  echo "running task !"
  exit 1
'''
```

That :point_up:  will output something like:

```bash
Retrying .. waiting 1000 ms before trying again ...
Trying task run_retry - try number 1
Retrying .. waiting 4000 ms before trying again ...
Trying task run_retry - try number 2
Retrying .. waiting 9000 ms before trying again ...
Trying task run_retry - try number 3
Max retries reached !
[cargo-make] ERROR - Error while running plugin: Source: Unknown Line: 29 - Exit with error code: 1
[cargo-make] WARN - Build Failed.
```

Let me know your thoughts @geofmureithi and @mariopil and if positive, i can extend this to rest of the contracts.

Reply:
Looks nice. So this will retry the task as long as its exit code != 0?

Reply:
Have you tried a delay?

Reply:
> Looks nice. So this will retry the task as long as its exit code != 0?

Yeah, its not perfect though, it would be nice to detect the `transaction timeout` specific error, so we would be able to differentiate among "retryable" and "non retryable" errors. But probably good for first iteration :stuck_out_tongue:

Reply:
> Have you tried a delay? This is possibly enough with the exponential backoff.

There is actually a delay , check the `sleep` command in code. Or you mean just a delay and a second retry would be enough ?